src/js/options.ts

```diff
@@ -20,6 +20,12 @@ export interface ReactNativeOptions extends BrowserOptions {
    */
   enableNativeCrashHandling?: boolean;
 
+  /** Initializes the native SDK on init.
+   * Set this to `false` if you have an existing native SDK and don't want to re-initialize.
+   * @default true
+   */
+  shouldInitializeNativeSdk?: boolean;
```

Comment:
what are the edge cases of such flag?

eg 1. RN SDK captures an event, Native SDKs are not inited.
2. RN sets up X DSN but events go to Y DSN etc...

not against it, but we'll need to figure out such edge cases and write them down.

Reply:
Yes I agree, should mention any possible side effects and edge cases, and also need to document this clearly. Thanks for the catch!

Reply:
Being the author of the initial feature request, I'd like to add my 2 cents: Having the only source for DSN/environment etc would be a better option, so that I don't need to worry about that in the JS side, since it'd've been already fully initialised and configured on the native side.

Reply:
@pioneer32 that would be the perfect solution, agree, but I'd say we'd go the other way around.
we'd sync the options from RN -> Native as we already do, which is let's say ~90+% of the use cases.
you could open an issue as a feature request and let's see if it gets upvoted, but I'd bet on the case described above.

Reply:
@marandaneto Thank you for clarification! My guess is `shouldInitializeNativeSdk` sort of going to disable calling https://github.com/getsentry/sentry-react-native/blob/97b16b72cc2970a9de89ff2903064a19879d2b8b/ios/RNSentry.m#L67 somehow (I might be wrong though). However, being keen to see what it'd look like, I commented out that line intentionally, and it seems to work as expected, almost :) FYI The problem I encountered was: all the tags/breadcrumbs/etc set to the native client are NOT copied into JS client before sending an error from JS.